File: chruch-cms/src/main/java/com/ephraim/chruch_cms/model/Event.java
package com.ephraim.chruch_cms.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.ToString;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Data
@Entity
@Table(name = "events")
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String title;

    private String description;

    @Column(nullable = false)
    private OffsetDateTime eventDateTime;

    private String location;

    @CreationTimestamp
    @Column(updatable = false)
    private OffsetDateTime createdAt;

    @UpdateTimestamp
    private OffsetDateTime updatedAt;

    @OneToMany(mappedBy = "event", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    @ToString.Exclude
    private List<Rsvp> rsvps = new ArrayList<>();

    @OneToMany(mappedBy = "event", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    @ToString.Exclude
    private List<Attendance> attendanceRecords = new ArrayList<>();
}

File: lib/models/event.dart
class Event {
  final int? id;
  final String title;
  final String? description;
  final String eventDate;
  final String? location;

  Event({
    this.id,
    required this.title,
    this.description,
    required this.eventDate,
    this.location,
  });

  factory Event.fromJson(Map<String, dynamic> j) => Event(
        id: j['id'],
        title: j['title'] ?? '',
        description: j['description'],
        eventDate: j['eventDate'] ?? '',
        location: j['location'],
      );

  Map<String, dynamic> toJson() => {
        'title': title,
        if (description != null) 'description': description,
        'eventDate': eventDate,
        if (location != null) 'location': location,
      };
}
